### PR TITLE
Remove operator-sdk and other unused package dependencies

### PR DESCRIPTION
**Pull request by pravisankar**
- operator-sdk pkg no longer used, needed functionality is provided by controller-runtime

- Remove unnecessary config dir

- Added or updated auto-generated defaults, deepcopy and openapi functions

- Added scripts to update and verify defaults,deepcopy,conversion and openapi funcs

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/sig-network-edge @ironcladlou PTAL

File: Gopkg.toml [hunk 1]
```diff
@@ -46,8 +40,6 @@ required = [
   name = "github.com/aws/aws-sdk-go"
   version = "v1.15.72"
 
-# Overrides for the operator-sdk
```
Comment (Miciah) [inline Gopkg.toml]: Does removing operator-sdk mean these overrides are no longer necessary?

File: manifests/02-deployment.yaml [hunk 2]
```diff
@@ -20,6 +20,7 @@ spec:
       containers:
         - name: ingress-operator
           image: openshift/origin-cluster-ingress-operator:latest
+          imagePullPolicy: IfNotPresent
```
Comment (Miciah) [inline manifests/02-deployment.yaml]: What is the reason for this change?

File: test/e2e/operator_test.go [hunk 3]
```diff
@@ -26,9 +25,9 @@ import (
 )
 
 func getClient() (client.Client, string, error) {
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		return nil, "", fmt.Errorf("failed to get watch namespace: %v", err)
+	namespace, ok := os.LookupEnv("WATCH_NAMESPACE")
+	if !ok {
+		return nil, "", fmt.Error("WATCH_NAMESPACE environment variable is required")
```
Comment (Miciah) [inline test/e2e/operator_test.go]: Need to use `fmt.Errorf`, or import `errors` and use `errors.New`.

**Comment by pravisankar** [comment]
@openshift/sig-network-edge updated, PTAL
